Update `PlatformPlugin` to not call `setStatusBarColor`, `setNavigationBarColor`, `setNavigationBarDividerColor` when disabled

AbdeMohlbi opened the pull request:
fixes #165582
towards #165327 (see https://discord.com/channels/608014603317936148/846507907876257822/1451187625116569660)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

**Note**: The Flutter team is currently trialing the use of [Gemini Code Assist for GitHub](https://developers.google.com/gemini-code-assist/docs/review-github-code). Comments from the `gemini-code-assist` bot should not be taken as authoritative feedback from the Flutter team. If you find its comments useful you can update your code accordingly, but if you are unsure or disagree with the feedback, please feel free to wait for a Flutter team member's review for guidance on which automated comments should be addressed.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

AbdeMohlbi committed:
Fix comments and update system UI color handling

AbdeMohlbi committed:
Merge branch 'master' into issue_165582

reidbaker commented:
Thank you for the work on this issue.

camsim99 commented on engine/src/flutter/shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:
```diff
@@ -477,7 +478,9 @@ private void setSystemChromeSystemUIOverlayStyle(
     }
 
     if (systemChromeStyle.statusBarColor != null) {
```
Nit: Maybe leave comments above the added if statements to explain why we aren't calling the methods below API level 35?

camsim99 reviewed (commented):
Looks good to me just some nits!

AbdeMohlbi committed:
reviewer's comment 1

Co-authored-by: Camille Simon <43054281+camsim99@users.noreply.github.com>

AbdeMohlbi committed:
reviewer's comment 2 about `new` and `WindowInsetsController`

Co-authored-by: Reid Baker <1063596+reidbaker@users.noreply.github.com>

AbdeMohlbi committed:
Clarify deprecation of system UI methods for Android 15+

Updated deprecation comments for system UI styling methods to clarify their ineffectiveness on Android 15 and above, and suggested using WindowInsetsController instead.

AbdeMohlbi committed:
the formatter should be happy now

AbdeMohlbi committed:
what the formatter try to prove here ?

Updated comment to clarify that setNavigationBarColor has no effect on Android 15 and above.

AbdeMohlbi committed:
Simplify comments on navigation bar divider color

Removed redundant comment lines regarding navigation bar divider color.

camsim99 reviewed (approved):
LGTM thanks!

camsim99 committed:
Merge branch 'master' into issue_165582

Permutoto commented:
@AbdeMohlbi Hi. It breaks app status bar coloring for apps that explicitly opts out from edge to edge :(
`activity.getTheme().applyStyle(R.style.OptOutEdgeToEdgeEnforcement, false);`

AbdeMohlbi commented:
@Permutoto please read https://github.com/flutter/flutter/issues/165582

Permutoto commented:
@AbdeMohlbi 
It states that those methods are 'disabled'. 
That's not entirely true. 
If the activity is opted out from edge to edge and apps targets api 35 then those methods work as before

AbdeMohlbi commented:
> @AbdeMohlbi It states that those methods are 'disabled'. That's not entirely true. If the activity is opted out from edge to edge and apps targets api 35 then those methods work as before

you can always file an issue with all your details/resources/repro steps and someone from the android team will take a look and decide about if it expected behavior or not.
